src/Microsoft.PowerShell.Commands.Management/commands/management/TestConnectionCommand.cs

```diff
@@ -0,0 +1,937 @@
+            record.RecordType = ProgressRecordType.Completed;
+            WriteProgress(record);
+        }
+
+#endregion MTUSizeTest
+
+#region PingTest
+
+        private void ProcessPing(String targetNameOrAddress)
+        {
+            String resolvedTargetName = null;
+            IPAddress targetAddress = null;
+
+            if (!InitProcessPing(targetNameOrAddress, out resolvedTargetName, out targetAddress))
+            {
+                return;
+            }
+
+            if (!Continues.IsPresent)
+            {
+                WritePingHeader(resolvedTargetName, targetAddress.ToString());
+            }
+
+            bool quietResult = true;
+            byte[] buffer = GetSendBuffer(BufferSize);
+
+            Ping sender = new Ping();
+            PingOptions pingOptions = new PingOptions(MaxHops, DontFragment.IsPresent);
+            PingReply reply = null;
+            PingReport pingReport = new PingReport(Source, resolvedTargetName);
+            Int32 timeout = TimeoutSeconds * 1000;
+            Int32 delay = Delay * 1000;
+
+            for (int i = 1; i <= Count; i++)
+            {
+                try
+                {
+                    reply = sender.Send(targetAddress, timeout, buffer, pingOptions);
+                }
+                catch (PingException ex)
+                {
+                    string message = StringUtil.Format(TestConnectionResources.NoPingResult, resolvedTargetName, ex.Message);
+                    Exception pingException = new System.Net.NetworkInformation.PingException(message, ex.InnerException);
+                    ErrorRecord errorRecord = new ErrorRecord(pingException,
+                                                              TestConnectionExceptionId,
+                                                              ErrorCategory.ResourceUnavailable,
+                                                              resolvedTargetName);
+                    WriteError(errorRecord);
+
+                    quietResult = false;
+                    continue;
+                }
+
+                if (Continues.IsPresent)
+                {
+                    WriteObject(reply);
+                }
+                else
+                {
+                    if (Quiet.IsPresent)
+                    {
+                        // Return 'true' only if all pings have completed successfully.
+                        quietResult &= reply.Status == IPStatus.Success;
+                    }
+                    else
+                    {
+                        pingReport.Replies.Add(reply);
+                    }
+
+                    WritePingProgress(reply);
+                }
+
+                // Delay between ping but not after last ping.
+                if (i < Count && Delay > 0)
+                {
+                    Thread.Sleep(delay);
+                }
+            }
+
+            if (!Continues.IsPresent)
+            {
+                WritePingFooter();
```

Comment:
maybe move this to a finally and always do this?

Reply:
I think we should break the feedback into two parts:
1. scripting functionality
2. interactive functionality (display output)

Now I have implemented the output to the screen as progress bar and as text (without ETS) - we have to decide what is best to use.

Also we could do a merge if scripting functionality is good and wait feedback for interactive functionality (display output).

As for the footer comment I haven't personal preference. I am waiting feedback how we better make the output to the screen for all options.